zomes/test/src/lib.rs

```diff
@@ -114,21 +176,34 @@ fn validate(data: ValidateData) -> ExternResult<ValidateCallbackResult> {
     let element = data.element.clone();
     let entry = element.into_inner().1;
     let entry = match entry {
-            ElementEntry::Present(e) => e,
-            _ => return Ok(ValidateCallbackResult::Valid),
+        ElementEntry::Present(e) => e,
+        _ => return Ok(ValidateCallbackResult::Valid),
     };
     if let Entry::Agent(_) = entry {
         match data.element.header().prev_header() {
             Some(header) => {
-                match must_get_valid_element(header.clone())?.signed_header().header() {
+                match must_get_valid_element(header.clone())?
+                    .signed_header()
+                    .header()
+                {
                     Header::AgentValidationPkg(pkg) => {
-                        return validate_joining_code(pkg.membrane_proof.clone())
-                    },
-                    _ => return Ok(ValidateCallbackResult::Invalid("No Agent Validation Pkg found".to_string()))
+                        if skip_proof() {
+                            return Ok(ValidateCallbackResult::Valid);
+                        }
+                        return validate_joining_code(pkg.membrane_proof.clone());
+                    }
+                    _ => {
+                        return Ok(ValidateCallbackResult::Invalid(
+                            "No Agent Validation Pkg found".to_string(),
+                        ))
+                    }
                 }
-            },
+            }
             None => {
-                return Ok(ValidateCallbackResult::Invalid("Impossible state".to_string()))}
+                return Ok(ValidateCallbackResult::Invalid(
+                    "Impossible state".to_string(),
```

Comment:
Do we want to be more specific here about why this state should be impossible? It's not immediately obvious to me.

```suggestion
                    ""Impossible state (entry being validated will always have a previous header)"".to_string(),
```